posthog/api/person.py

```diff
@@ -237,14 +238,28 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> response.
         elif not filter.limit:
             filter = filter.shallow_clone({LIMIT: DEFAULT_PAGE_LIMIT})
 
-        query, params = PersonQuery(filter, team.pk).get_query(paginate=True, filter_future_persons=True)
-
-        raw_result = insight_sync_execute(
-            query, {**params, **filter.hogql_context.values}, filter=filter, query_type="person_list", team_id=team.pk
+        person_query = PersonQuery(filter, team.pk)
+        paginated_query, paginated_params = person_query.get_query(paginate=True, filter_future_persons=True)
+        raw_paginated_result = insight_sync_execute(
+            paginated_query,
+            {**paginated_params, **filter.hogql_context.values},
+            filter=filter,
+            query_type="person_list",
+            team_id=team.pk,
         )
-
-        actor_ids = [row[0] for row in raw_result]
-        actors, serialized_actors = get_people(team, actor_ids)
+        actor_ids = [row[0] for row in raw_paginated_result]
+        _, serialized_actors = get_people(team, actor_ids)
+
+        total_query, total_params = person_query.get_query(paginate=False, filter_future_persons=True)
```

Comment:
Not something we need to fix now, but the persons endpoint and how we paginate here feels very bespoke. Would be great to have the right abstractions in place for a common thing like pagination.

Reply:
Yeah, though here this gets a bit tricky because its ClickHouse – no ORM involved